### ISSUE TITLE
Is there any useful tooling available to pin library contracts without going fully typed (e.g., TypeScript)?

### ISSUE THREAD

**Issue by trevtrich** [cypress-io/cypress-example-recipes#373]
**TL;DR**
Is there anything out there currently to support something like types (TypeScript or whatever it may be) at the boundary of an app? So I wouldn't have to use TypeScript within my app, but could use it purely at the boundary.

To limit the problem set here, lets assume I own the library I want to build these types for.

**Detail**
This original question stemmed from a Twitter thread recently where I asked how one could pin a contract similar to how Pact pins an http contract but do so for a library. For the original thread see [this](https://twitter.com/IntelXDesign/status/1197615580657774592).

An example scenario (real-world case for the team I'm on) would be something like using a google-cloud-storage sdk. The GCS library is a client library that behind the scenes uses Google's public http apis. I could technically create a contract (using VCR or similar) by letting a live call go through the library where it then hits the true http apis. That would work. However, I would strongly prefer to not have to update my tests when Google changes the usage of their own apis.

The route we currently implement is to stub the library out entirely. This works just fine to keep tests fast and not brittle, but the obvious problem is that when Google does release a breaking change to the library calls there is no way to be made aware of that with tests because our tests mock it out.
